### DISCUSSION TITLE
State of power_limiter branch

### DISCUSSION THREAD

**Discussion by helgeerbe** [hoylabs/OpenDTU-OnBattery#107] [General]
Hi @berni2288,
what is the actual state of your branch? I'm eager to pull it into development.

**Comment by berni2288** [comment]
Hi @helgeerbe 
feel free to do so :) It should be stable enough.

The pin config for the Pylontech CAN Bus is still done with constants however, and not via the OpenDTU pin manager, but this shouldn't block the merge.
And the README.md maybe needs to be adjusted, since it contains infos from the old repo.
Feel free to edapt the README.md in the powerlimiter branch before merging it.

**Comment by berni2288** [comment]
I'm planning to do more improvements:

- More settings for Powerlimiter
- Code cleanup & Code comments
- Lots of documentation on how to connect the battery etc.

But not sure when I have time.

**Comment by helgeerbe** [comment]
OK, what I see so far. 

You hardcoded for the power_limit to the first inverter and the first channel? I have two inverters running. This needs to be configurable items.

You changed MQTT Settings and subscribes to the power limit settings. But you didn't assign any callback functions. I will remove that.

**Reply by berni2288** [reply]
> You hardcoded for the power_limit to the first inverter and the first channel? I have two inverters running. This needs to be configurable items.

Yes only 1 inverter (the first) is supported at the moment. Making the inverter configurable should be not so hard. However adjusting the limit of multiple inverters requires a bit more work. I only have 1 inverter.

> 
> You changed MQTT Settings and subscribes to the power limit settings. But you didn't assign any callback functions. I will remove that.

What? I don't know what you are talking about. Can you send me a link to the code line please?

**Comment by berni2288** [comment]
I've created a PR now to enable a better discussion: https://github.com/helgeerbe/OpenDTU-OnBattery/pull/108

**Reply by helgeerbe** [reply]
But I merged already in my local branch.

**Comment by helgeerbe** [comment]
Merge is done to development:

To Dos
- inverter is now set to 1 (must be configurable)
- channel is set to ch0 (must be configurable)

**Comment by spcqike** [comment]
is it possible to use the battery version without power limit / mqtt for a meter?

i would like to produce 200W if my battery is within its specs and cut off / start again. but somehow i cant find this in the firmware. it always expect me to set a mqtt topic for a meter (which makes sense if you want to dynamicly increase / decrease your production, depending on your consumption)

maybe i just cnt find it (i'm not used to search firmwares and binarys in the action tab :D)

**Reply by helgeerbe** [reply]
Actual this is not possible to disable the power meter.  Do you use home assistant or node-red. In the past I just used the load flag from the MPPT. I set the power limit manually to 150 W. If load is on, I turned inverter on, if load flag turns to off, I turned it off. It's a simple rule in node red or automation in home assistant.

**Reply by spcqike** [reply]
Yes. I use node red and MQTT at home. Right now I try to test (discharge test) a battery pack at work. I don’t want to transport the batteries if they are useless :) 

I think I will find an external solution to not discharge the battery while testing them :) 

anyway, can you tell me why there are three MQTT meter inputs? Does opendtu expect one input per phase? Right now I only get the total usage from my meter.

**Reply by helgeerbe** [reply]
The Shelly 3em is a 3 phase power meter. At the end it is just the sum of 3 phases. So one number on a single topic should be fine in your case.

**Reply by spcqike** [reply]
I just asked because it looked like he expected a topic in each input field. 

If I keep the battery, I will try at home with real MQTT meter reading, again :) 

TY so far :)

**Reply by berni2288** [reply]
Yes that's right, you can only use one of the 3 MQTT power meter settings. If you don't have a Powermeter but an MQTT server, you could also just create a static topic for now with a custom Watt value.

I'll implement an option to be able to disable the Powermeter reading, shouldn't be too hard.
